source/common/network/cidr_range.cc

```diff
@@ -79,13 +79,20 @@ IpVersion CidrRange::version() const {
 }
 
 bool CidrRange::isInRange(InstanceConstSharedPtr address) const {
-  if (address == nullptr || length_ < 0 || address->type() != Type::Ip ||
-      address->ip()->version() != version()) {
+  if (address == nullptr) {
+    return false;
+  }
+
+  return isInRange(*address);
+}
+
+bool CidrRange::isInRange(const Instance& address) const {
+  if (length_ < 0 || address.type() != Type::Ip || address.ip()->version() != version()) {
     return false;
   }
   // Make an CidrRange from the address, of the same length as this. If the two ranges have
   // are the same, then the address is in this range.
-  CidrRange other = create(address, length_);
+  CidrRange other = create(address.ip()->addressAsString(), length_);
```

Comment:
Sure, whichever makes sense, but it seems we should be able to do better than converting to/from a string for an operation that is essentially about copying an address and masking some bits. Of course, given the `friendly_address_` representation in string form that is maintained today, this might not be that much more efficient, but at least from a logical point of view we aren't mucking around with strings.